### PR TITLE
Add documentation for new type hints in Koalas

### PR DESCRIPTION
**Pull request by HyukjinKwon**
This PR adds some documentations for https://github.com/databricks/koalas/pull/1578

Screen Shot 2020-06-16 at 7 07 34 PM
Screen Shot 2020-06-16 at 7 07 50 PM

### CODE REVIEW

File: docs/source/user_guide/typehints.rst [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+.. currentmodule:: databricks.koalas
+
+Koalas, by default, infers the schema by taking some top records from the output,
+in particular, when you use APIs that allow users to apply a function against Koalas DataFrame
+such as :func:`DataFrame.transform`, :func:`DataFrame.apply`, :func:`DataFrame.transform_batch`,
+:func:`DataFrame.apply_batch`, :func:`Series.transform_batch`, etc.
+
+However, this is potentially expensive. If there are several expensive operations such as a shuffle
+in the upstream of the execution plan, Koalas will end up with executing twice, once for schema
```
Comment (itholic) [inline docs/source/user_guide/typehints.rst:13]: How about `Koalas will end up with executing twice` -> `Koalas will end up with executing Spark job twice` ?

File: docs/source/user_guide/typehints.rst [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+-------------------------------------
+
+In the early Koalas version, it was introduced to specify a type hint in the function in order to use
+it as a Spark schema. As an example, you can specify the return type hint as below by using Koalas
+:class:`DataFrame`.
+
+.. code-block:: python
+
+    >>> def pandas_div(pdf) -> ks.DataFrame[float, float]:
+    ...    # pdf is a pandas DataFrame,
```
Comment (itholic) [inline docs/source/user_guide/typehints.rst:34]: nit: `,` -> `.`

**Comment by itholic** [comment]
Looks really good to me !
(And also, how about we merge this and #1589 right after 1.0.0 will be actually released ??)

**Comment by HyukjinKwon** [comment]
The documentation should be the part of the release so it should include.

**Comment by itholic** [comment]
@HyukjinKwon Ah, makes sense.

**Comment by HyukjinKwon** [comment]
I am merging this to unblock the release.

**Pull request merged**